### ISSUE TITLE
Plugin very unstable since latest updates: Login and/or portal sync failed

### ISSUE THREAD

**Issue by jastorino82** [mrjackyliang/homebridge-adt-pulse#86]
### Describe the bug:
Since the huge Pulse outage and the latest 2.1.4 release of the plugin, the HomeKit integration is pretty unusable. Here is what I experience

After updating to 2.1.4 everything works great, but only very temporarily.  Sometimes it lasts a few hours only, other times slightly longer.  After a relatively short period of time, what seems to be happening is that the plugin is logged out from ADT.  After that, it will fail to login again, or it will login again but portal sync will fail per the below logs. 

Rebooting HomeBridge or shutting down / and starting the plugin container fixes the issue, but it is now very short lived. The Pulse app continues to work fine. Sometimes, it fixes itself some time later, but then goes unstable again shortly after.

In summary, the plugin at this moment is behaving very unstable.

### Steps to reproduce the bug:
Update to v2.1.4, wait 4-12 hours, then try to control the security panel from either the HomeBridge plugin or HomeKit

### Expected behavior:
Pulse portal sync should consistently be syncing every few seconds without the need to constantly reboot HomeBridge or the plugin container

### The Apple device you are using:
- Device: iPhone/iPad
- OS: iOS/iPadOS 15.6

### Homebridge information:
Find these 4 lines in the beginning of the logs, then replace the text inside the grave accents:

N/A

### Logs related to this bug:
```sh
###EXAMPLE OF LOGIN FAIL
ADT Pulse: Performing portal sync...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Portal sync failed.
ADT Pulse: Logging in...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Web portal version -> 23.0.0-99
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp
ADT Pulse: Response path matches -> false
ADT Pulse: Login failed.
[10/08/2022, 10:58:36] [ADT Pulse] Login failed. Trying again...
[10/08/2022, 10:58:36] [ADT Pulse] Sign In unsuccessful. Try again. Check your Caps Lock key.

#EXAMPLE OF LOGIN SUCCESS BUT SYNC FAIL
[10/08/2022, 13:51:34] [ADT Pulse] Setting Security Panel (system-1) status from stay to off...
ADT Pulse: Setting device status...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Set device status to off failed.
[10/08/2022, 13:51:35] [ADT Pulse] Set device status failed.
[10/08/2022, 13:51:35] [ADT Pulse] You have not yet signed in or you have been signed out due to inactivity. Sign in to proceed.
ADT Pulse: Logging in...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Web portal version -> 23.0.0-99
ADT Pulse: Response path -> /myhome/23.0.0-99/summary/summary.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Site ID -> xxxxxxxxxx
ADT Pulse: Login success.
ADT Pulse: Performing portal sync...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Portal sync failed.

#EXAMPLE OF FAILURE, AND IT FIXES ITSELF BUT NOT FOR 20 MINUTES. NOTE TIMESTAMPS
[10/08/2022, 13:51:34] [ADT Pulse] Setting Security Panel (system-1) status from stay to off...
ADT Pulse: Setting device status...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Set device status to off failed.
[10/08/2022, 13:51:35] [ADT Pulse] Set device status failed.
[10/08/2022, 13:51:35] [ADT Pulse] You have not yet signed in or you have been signed out due to inactivity. Sign in to proceed.
ADT Pulse: Logging in...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Web portal version -> 23.0.0-99
ADT Pulse: Response path -> /myhome/23.0.0-99/summary/summary.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Site ID -> xxxxxxxxxx
ADT Pulse: Login success.
ADT Pulse: Performing portal sync...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Portal sync failed.
ADT Pulse: Logging in...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Web portal version -> 23.0.0-99
ADT Pulse: Response path -> /myhome/23.0.0-99/summary/summary.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Site ID -> xxxxxxxxxx
ADT Pulse: Login success.
ADT Pulse: Performing portal sync...
ADT Pulse: Response path -> /myhome/23.0.0-99/Ajax/SyncCheckServ?t=1660154922504
ADT Pulse: Response path matches -> true
ADT Pulse: Portal sync success.
ADT Pulse: Getting device status...
ADT Pulse: Response path -> /myhome/23.0.0-99/ajax/orb.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Get device status success.
[10/08/2022, 14:08:43] [ADT Pulse] Security Panel (system-1) changed from "armed stay / all quiet" to "disarmed / all quiet".
ADT Pulse: Getting zone status...
ADT Pulse: Response path -> /myhome/23.0.0-99/ajax/orb.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Get zone status success.

#40 MINUTES AFTER IT FIXES ITSELF, IT FAILS AGAIN...
[10/08/2022, 14:08:43] [ADT Pulse] Security Panel (system-1) changed from "armed stay / all quiet" to "disarmed / all quiet".
ADT Pulse: Getting zone status...
ADT Pulse: Response path -> /myhome/23.0.0-99/ajax/orb.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Get zone status success.
ADT Pulse: Performing portal sync...
ADT Pulse: Response path -> /myhome/23.0.0-99/Ajax/SyncCheckServ?t=1660155519145
ADT Pulse: Response path matches -> true
ADT Pulse: Portal sync success.
ADT Pulse: Getting device status...
ADT Pulse: Response path -> /myhome/23.0.0-99/ajax/orb.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Get device status success.
ADT Pulse: Getting zone status...
ADT Pulse: Response path -> /myhome/23.0.0-99/ajax/orb.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Get zone status success.
ADT Pulse: Performing portal sync...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Portal sync failed.
ADT Pulse: Logging in...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Web portal version -> 23.0.0-99
ADT Pulse: Response path -> /myhome/23.0.0-99/summary/summary.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Site ID -> xxxxxxxxxx
ADT Pulse: Login success.
ADT Pulse: Performing portal sync...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Portal sync failed.
ADT Pulse: Logging in...
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp
ADT Pulse: Response path matches -> true
ADT Pulse: Web portal version -> 23.0.0-99
ADT Pulse: Response path -> /myhome/23.0.0-99/access/signin.jsp?e=ns&partner=adt
ADT Pulse: Response path matches -> false
ADT Pulse: Login failed.
[10/08/2022, 14:48:41] [ADT Pulse] Login failed. Trying again...
[10/08/2022, 14:48:41] [ADT Pulse] You have not yet signed in or you have been signed out due to inactivity. Sign in to proceed.
```

**Comment by mrjackyliang** [comment]
You can try to downgrade to `v2.1.3` but as far as I know so far, `v2.1.4` changes were just the user agent.

I understand your frustration, but blaming it entirely on my plugin is wrong.

Mind you that ADT Pulse is an old system, and Alarm.com will not be making better changes towards this platform other than causing a couple of problems here and there to make you annoyed so you either upgrade to the latest ADT Control system or move out of ADT entirely and use Alarm Decoder/EnvisaLink.

**Comment by jastorino82** [comment]
Hi,

Please don't misunderstand.  I love the plugin and appreciate all the work you put into maintaining it for the community.  Sorry if my bug report came off the wrong way.  I'm not trying to say it's all your fault, just providing the information best I can. Cheers.

**Comment by jastorino82** [comment]
Good news, everything has been stable now for about 24 hours.  For others that might be facing this, here is what I did to fix the issue - I simply logged into the Pulse portal, and deleted all my saved devices.  Then, I went through the process in the readme again to establish a new saved device and MFA fingerprint for HomeBridge.  I updated my plugin configuration with the new MFA fingerprint and reboot, and it has been solid ever since.
